### PR TITLE
fix: add optional buttons to delete function

### PR DESCRIPTION
**Pull request by andrewKapustin22**

### CODE REVIEW

File: scripts/secret.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+LwRe/IK+TIyPANAUICfYpYoQ5sz7x0zdtxoxoY0J6oE=
```
Comment (MikhailSadovsky) [inline scripts/secret.txt:1]: Please don't commit this file

File: components/scg/static/components/html/demo-scg-delete-panel.html [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<div class="demo-fragments-window">
```
Comment (MikhailSadovsky) [inline components/scg/static/components/html/demo-scg-delete-panel.html:1]: Please don't use 'demo' name

File: client/static/components/html/demo-scg-delete-panel.html [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<div class="demo-fragments-window">
```
Comment (MikhailSadovsky) [inline client/static/components/html/demo-scg-delete-panel.html:1]: Please don't commit this file, include in gitignore

**Review by MikhailSadovsky — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 4]
```diff
@@ -82,3 +82,8 @@ build_kb.sh
 .idea/
 
 sc-server*
+
+scripts/secret.txt
+
+client/static/components/html
+
```
Comment (MksmOrlov) [inline .gitignore:89]: add newline please

File: components/scg/static/components/html/scg-delete-panel.html
Comment (MksmOrlov) [inline components/scg/static/components/html/scg-delete-panel.html:1]: And what about English mode? There is only Russian text

File: .gitignore (hunk 4 above)
Comment (MksmOrlov) [reply .gitignore:89]: just remove space in the last line

**Review by MksmOrlov — APPROVED** [review]

File: client/js/Ui/core.js [hunk 5]
```diff
@@ -28,8 +28,9 @@ SCWeb.ui.Core = {
               SCWeb.ui.WindowManager.init(data),
               SCWeb.ui.SearchPanel.init(),
               SCWeb.ui.ExpertModePanel.init(),
+            //   SCWeb.ui.DeleteScgButtons.init(),
```
Comment (MikhailSadovsky) [inline client/js/Ui/core.js:31]: Please don't leave commented code

File: client/js/Ui/deleteScgButtons.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
+SCWeb.ui.DeleteScgButtons = {
+
+    init: function() {
+        return new Promise(resolve => {
+            var expert_mode_identifier = 'ui_scg_control_tool_delete_from_working_place';
+            this.expert_mode_container_id = '#' + 'delete_buttons_container';
+            var self = this;
+
+            SCWeb.core.Server.resolveScAddr([expert_mode_identifier]).then(function (addrs) {
```
Comment (MikhailSadovsky) [inline client/js/Ui/deleteScgButtons.js:9]: Please don't use `expert mode` words

File: client/js/Ui/deleteScgButtons.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
+SCWeb.ui.DeleteScgButtons = {
```
Comment (MikhailSadovsky) [inline client/js/Ui/deleteScgButtons.js:1]: Is this file really needed?

File: components/scg/src/deleteScgButtons.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    init: function() {
+        return new Promise(resolve => {
+            var delete_from_working_place_button_identifier = 'ui_scg_control_tool_delete_from_working_place';
+            var delete_from_db_button_identifier = 'ui_scg_control_tool_delete_from_knowledge_base';
+            this.delete_buttons_container_id = '#' + 'delete_buttons_container';
+
+            var self = this;
+
+            SCWeb.core.Server.resolveScAddr([delete_from_working_place_button_identifier, delete_from_db_button_identifier]).then(function (addrs) {
+                var delete_from_wp = addrs[delete_from_working_place_button_identifier];
```
Comment (MikhailSadovsky) [inline components/scg/src/deleteScgButtons.js:12]: Please rename wp & dp to more clear names

File: components/scg/src/deleteScgButtons.js [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+                        SCWeb.core.EventManager.subscribe("translation/get", self, function (objects) {
+                            $(self.delete_buttons_container_id + ' [sc_addr]').each(function (index, element) {
+                                objects.push($(element).attr('sc_addr'));
+                            });
+                        });
+                        resolve();
+                    });
+                }
+
+                if(delete_from_db) {
```
Comment (MikhailSadovsky) [inline components/scg/src/deleteScgButtons.js:30]:
```suggestion
                if (delete_from_db) {
```

File: components/scg/src/scg.js [hunk 10]
```diff
@@ -117,8 +117,25 @@ SCg.Editor.prototype = {
...
+                                error: function () {
+                                    SCgDebug.error(
+                                        "Error to get delete panel");
+                                },
+                                complete: function () {
+                                    self.bindToolEvents();
+                                }
+                            })
                         }
+                        // complete: function () {
```
Comment (MikhailSadovsky) [inline components/scg/src/scg.js:136]: Please remove unused code

File: components/scg/src/scg.js [hunk 11]
```diff
@@ -117,8 +117,25 @@ SCg.Editor.prototype = {
                                 "Error to get edges type change panel");
                         },
                         complete: function () {
-                            self.bindToolEvents();
+                            $.ajax({
+                                url: 'static/components/html/scg-delete-panel.html',
+                                dataType: 'html',
+                                success: function (response) {
+                                    // DeleteButtons.init(),
```
Comment (MikhailSadovsky) [inline components/scg/src/scg.js:124]: Please remove unused code

File: components/scg/src/scg.js [hunk 12]
```diff
@@ -690,30 +707,124 @@ SCg.Editor.prototype = {
...
+                        self.scene.addDeletedObjects(deletableObjects);
+                    } else {
+                        self.scene.deleteObjects(self.scene.selected_objects);
+                        self.scene.addDeletedObjects(self.scene.selected_objects);
+                    }
+                }
+                self.hideTool(self.toolDelete())
+                stop_modal();
+                select.button('toggle');
+                // if (self.scene.selected_objects.length > 1) {
```
Comment (MikhailSadovsky) [inline components/scg/src/scg.js:794]: Please remove unused code

**Review by MikhailSadovsky — CHANGES_REQUESTED** [review]

File: components/scg/src/scg.js [hunk 13]
```diff
@@ -690,30 +703,99 @@ SCg.Editor.prototype = {
...
+                    show: 500,
+                    hide: 100
+                }
+            }).popover('show');
+            cont.find('.popover-content').append(
+                '<button id="scg-close-btn" type="button" class="close scg-close-btn-fragments-window">&times;</button>'
+            );
+
+            if (self.scene.selected_objects.length === 1) {
+                if(!self.scene.selected_objects[0].sc_addr) {
```
Comment (MikhailSadovsky) [inline components/scg/src/scg.js:735]: Please fix **everywhere** in file
```suggestion
                if (!self.scene.selected_objects[0].sc_addr) {
```

**Review by MikhailSadovsky — APPROVED** [review]

**Pull request merged**